extensions/import/src/wizard/pages/deployPlanPage.ts

```diff
@@ -0,0 +1,288 @@
+						horizontal: true
+					},
+					title: ''
+				},
+				dataLossComponent
+			]
+		};
+	}
+
+	private getTableColumns(dataloss: boolean): sqlops.TableColumn[] {
+		let columns: sqlops.TableColumn[] = [
+			{
+				value: localize('dacfx.operationColumn', 'Operation'),
+				width: 75,
+				toolTip: localize('dacfx.operationTooltip', 'Operation that will occur during deployment')
+			},
+			{
+				value: localize('dacfx.typeColumn', 'Type'),
+				width: 100,
+				toolTip: localize('dacfx.typeTooltip', 'Type of object')
+			},
+			{
+				value: localize('dacfx.objectColumn', 'Object'),
+				width: 300,
+				toolTip: localize('dacfx.objecTooltip', 'Object name')
+			}];
+
+		if (dataloss) {
+			columns.unshift(
+				{
+					value: localize('dacfx.dataLossColumn', 'Data Loss'),
+					width: 50,
+					cssClass: 'center-align',
+					toolTip: localize('dacfx.dataLossTooltip', 'Operations that may result in data loss are marked with a warning sign')
+				});
+		}
+		return columns;
+	}
+
+	private parseXml(report: string): Result {
+		let operations = [];
+		let dataLossAlerts = new Set<string>();
+
+		let currentOperation = '';
+		let dataIssueAlert = false;
+		let currentReportSection: deployPlanXmlTag;
+		let currentTableObj: TableObject;
+		let p = new parser.Parser({
+			onopentagname(name) {
+				if (name === deployPlanXmlTag.Alert) {
+					currentReportSection = deployPlanXmlTag.Alert;
+				} else if (name === deployPlanXmlTag.Operation) {
+					currentReportSection = deployPlanXmlTag.Operation;
+				} else if (name === deployPlanXmlTag.Item) {
+					currentTableObj = new TableObject();
+				}
+			},
+			onattribute: function (name, value) {
+				if (currentReportSection === deployPlanXmlTag.Alert) {
+					switch (name) {
+						case attributeName.Name: {
+							// only care about showing data loss alerts
+							if (value === attributeName.DataIssue) {
+								dataIssueAlert = true;
+							}
+							break;
+						}
+						case attributeName.Id: {
+							if (dataIssueAlert) {
+								dataLossAlerts.add(value);
+							}
+							break;
+						}
+					}
+				} else if (currentReportSection === deployPlanXmlTag.Operation) {
+					switch (name) {
+						case attributeName.Name: {
+							currentOperation = value;
+							break;
+						}
+						case attributeName.Value: {
+							currentTableObj.object = value;
+							break;
+						}
+						case attributeName.Type: {
+							currentTableObj.type = value;
+							break;
+						}
+						case attributeName.Id: {
+							if (dataLossAlerts.has(value)) {
+								currentTableObj.dataloss = true;
+							}
+							break;
+						}
+					}
+				}
+			},
+			onclosetag: function (name) {
+				if (name === deployPlanXmlTag.Item) {
+					currentTableObj.operation = currentOperation;
+					operations.push(currentTableObj);
+				}
+			}
+		}, { xmlMode: true, decodeEntities: true });
+		p.parseChunk(report);
+
+		let data = [];
+		operations.forEach(operation => {
+			if (dataLossAlerts.size > 0) {
```

Comment:
Would it simplify things to always add the isDataLoss column, but remove/hide it later when it's time to render it?

Reply:
I think it would be the same since to remove the column, the array needs to be looped through again to remove it from each entry

Reply:
Sure, but then we have the logic in one place.